internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs

```diff
@@ -1215,10 +1234,161 @@ mod tests {
             .unwrap();
 
         repository
-            .remove_rolled_back_transactions_and_block_range(BlockRange::LENGTH * 3)
+            .remove_rolled_back_transactions_and_block_range_by_block_number(BlockRange::LENGTH * 3)
             .await
             .unwrap();
         assert_eq!(2, repository.get_all_transactions().await.unwrap().len());
         assert_eq!(2, repository.get_all_block_range_root().unwrap().len());
     }
+
+    #[tokio::test]
+    async fn remove_rolled_back_transactions_and_block_range_by_slot_number() {
+        let repository = CardanoTransactionRepository::new(Arc::new(
+            SqliteConnectionPool::build(1, cardano_tx_db_connection).unwrap(),
+        ));
+
+        repository
+            .create_transactions(vec![
+                CardanoTransactionRecord::new(
+                    "tx_hash-123",
+                    BlockNumber(10),
+                    SlotNumber(50),
+                    "block_hash-111",
+                ),
+                CardanoTransactionRecord::new(
+                    "tx_hash-456",
+                    BlockNumber(11),
+                    SlotNumber(51),
+                    "block_hash-000",
+                ),
+                CardanoTransactionRecord::new(
+                    "tx_hash-789",
+                    BlockNumber(13),
+                    SlotNumber(52),
+                    "block_hash-333",
+                ),
+                CardanoTransactionRecord::new(
+                    "tx_hash-000",
+                    BlockNumber(101),
+                    SlotNumber(100),
+                    "block_hash-222",
+                ),
+            ])
+            .await
+            .unwrap();
+
+        {
+            repository
+                .remove_rolled_back_transactions_and_block_range_by_block_number(BlockNumber(110))
+                .await
+                .expect("Failed to remove rolled back transactions");
+
+            let transactions = repository
+                .get_all()
+                .await
+                .unwrap()
+                .into_iter()
+                .map(|v| v.into())
+                .collect::<Vec<_>>();
+            assert_eq!(
+                vec![
```

Comment:
The creation of one transaction in 5 lines does not allow to see all the test on the screen.
We may create a local function to hide useless information and create transaction in one line.
We can obtain something like:
```
repository
            .create_transactions(vec![
                transaction_record(BlockNumber(10), SlotNumber(50)),
                transaction_record(BlockNumber(11), SlotNumber(51)),
                transaction_record(BlockNumber(13), SlotNumber(52)),
                transaction_record(BlockNumber(101), SlotNumber(100)),
            ])
            .await
            .unwrap();
```

Reply:
Done :heavy_check_mark: